src/libcore/cmp.rs

```diff
@@ -1208,6 +1208,36 @@ mod impls {
         #[inline]
         fn gt(&self, other: &&mut B) -> bool { PartialOrd::gt(*self, *other) }
     }
+    #[stable(feature = "partial_ord_mut_ref", since = "1.40.0")]
+    impl<A: ?Sized, B: ?Sized> PartialOrd<&B> for &mut A where A: PartialOrd<B> {
+        #[inline]
+        fn partial_cmp(&self, other: &&B) -> Option<Ordering> {
+            PartialOrd::partial_cmp(*self, *other)
+        }
+        #[inline]
+        fn lt(&self, other: &&B) -> bool { PartialOrd::lt(*self, *other) }
+        #[inline]
+        fn le(&self, other: &&B) -> bool { PartialOrd::le(*self, *other) }
+        #[inline]
+        fn ge(&self, other: &&B) -> bool { PartialOrd::ge(*self, *other) }
+        #[inline]
+        fn gt(&self, other: &&B) -> bool { PartialOrd::gt(*self, *other) }
+    }
+    #[stable(feature = "partial_ord_mut_ref", since = "1.40.0")]
+    impl<A: ?Sized, B: ?Sized> PartialOrd<&mut B> for &A where A: PartialOrd<B> {
+        #[inline]
+        fn partial_cmp(&self, other: &&B) -> Option<Ordering> {
```

Comment:
```suggestion
        fn partial_cmp(&self, other: &&mut B) -> Option<Ordering> {
```

and the same for the methods below.